### PR TITLE
Bandannas and Goggles

### PR DESCRIPTION
**Pull request by Metroidlover**
## About The Pull Request

Adds Various bandannas as mask items capable of standing in for Rebreathers and such while being able to be pulled down to remove their effects. Also renames every instance of "bandana" to "bandanna"
Also Adds new sprites for Goggles and puts both in Marine Vendors

_Images to be added upon request_

## Why It's Good For The Game

New cool looking things along with spelling

## Changelog
:cl: Kuro & MetroidLover
add: Added Bandanas of all Squad and Map colors to be worn on your face. Also yes, there is a skull one
Add: Ballistic Goggles and the new Bandannas are now able to be found in Marine Surplus Vendors
spellcheck: "Bandanna" instead of "Bandana"
imageadd: New sprites for Ballistic Goggles
/:cl:

### CODE REVIEW

**Comment by Pariah919** [comment]
speedmerge

**Comment by Kuro-ai** [comment]
oh yeah it's bandanna time

**Comment by Spyroshark** [comment]
SPEEDMERGE SO I CAN WEAR BANDANNA PLEASE

File: code/modules/clothing/masks/breath.dm [hunk 1]
```diff
@@ -62,7 +62,68 @@
...
+	name = "purple bandanna"
+	desc = "A colored, resilient, and insulating cloth to cover your face from the elements. This one is colored Grape Purple."
+	icon_state = "charlie_bandanna"
+	item_state = "charlie_bandanna"
+
+/obj/item/clothing/mask/rebreather/bandanna/delta
+	name = "blue bandanna"
+	desc = "A colored, resilient, and insulating cloth to cover your face from the elements. This one is colored Blueberry Blue."
+	icon_state = "delta_bandanna"
+	item_state = "delta_bandanna"
```
Comment (Rohesie) [inline code/modules/clothing/masks/breath.dm]:
```suggestion
	item_state = "delta_bandanna"

```
Try to end files in a new line.

File: code/modules/clothing/masks/breath.dm [hunk 2]
```diff
@@ -62,7 +62,68 @@
 	desc = "A close-fitting cap that covers the top, back, and sides of the head. Can also be adjusted to cover the lower part of the face so it keeps the user warm in harsh conditions."
 	icon_state = "coif"
 	item_state = "coif"
-	flags_inventory = COVERMOUTH|ALLOWREBREATH
 	flags_inv_hide = HIDEALLHAIR|HIDEEARS
 	flags_cold_protection = HEAD
 	min_cold_protection_temperature = ICE_PLANET_MIN_COLD_PROTECTION_TEMPERATURE
+
+/obj/item/clothing/mask/rebreather/bandanna
```
Comment (Rohesie) [inline code/modules/clothing/masks/breath.dm]: Why is this a child of the `rebreather`?

File: code/modules/clothing/masks/breath.dm [hunk 3]
```diff
@@ -62,7 +62,68 @@
...
+	name = "bandanna"
+	desc = "A colored, resilient, and insulating cloth to cover your face from the elements."
+	icon_state = "bandanna"
+	item_state = "bandanna"
+	flags_inv_hide = HIDEFACE
+	flags_armor_protection = FACE
+
+	verb/toggle()
```
Comment (Rohesie) [inline code/modules/clothing/masks/breath.dm]:
```suggestion
/obj/item/clothing/mask/rebreather/bandanna/verb/toggle()
```
I understand this is a copypaste, but always use full pathing.

File: code/modules/clothing/masks/breath.dm [hunk 4]
```diff
@@ -62,7 +62,68 @@
...
+		set category = "Object"
+		set name = "Adjust bandanna"
+		set src in usr
+
+		if(usr.canmove && !usr.stat && !usr.restrained())
```
Comment (Rohesie) [inline code/modules/clothing/masks/breath.dm]:
```suggestion
		if(incapacitated())
			return
```
Use early returns.

File: code/modules/clothing/masks/breath.dm [hunk 5]
```diff
@@ -62,7 +62,68 @@
...
+			active = !active
+			icon_state = "[initial(icon_state)][!active ? "_down" : ""]"
+			if(!active)
+				DISABLE_BITFIELD(flags_inventory, COVERMOUTH|ALLOWREBREATH)
```
Comment (Rohesie) [inline code/modules/clothing/masks/breath.dm]:
```suggestion
				DISABLE_BITFIELD(flags_inventory, (COVERMOUTH|ALLOWREBREATH))
```
Get those bitflag defines in a nice `()` embrace.

File: code/modules/clothing/masks/breath.dm [hunk 6]
```diff
@@ -62,7 +62,68 @@
...
+				DISABLE_BITFIELD(flags_inv_hide, HIDEFACE|HIDELOWHAIR)
+				DISABLE_BITFIELD(flags_armor_protection, FACE)
+			else
+				ENABLE_BITFIELD(flags_inventory, COVERMOUTH|ALLOWREBREATH)
+				ENABLE_BITFIELD(flags_inv_hide, HIDEFACE|HIDELOWHAIR)
+				ENABLE_BITFIELD(flags_armor_protection, FACE)
+			if(usr)
```
Comment (Rohesie) [inline code/modules/clothing/masks/breath.dm]: Can this be called without a `usr`?

File: code/modules/clothing/masks/breath.dm (hunk 2 above)
Comment (Metroidlover) [reply code/modules/clothing/masks/breath.dm]: I mean, should it NOT be? I did it because the Rebreather has all the same flags I want

Comment (spookydonut) [reply code/modules/clothing/masks/breath.dm]: Yeah that's not how that works

**Comment by spookydonut** [comment]
Needs work.

File: code/modules/clothing/masks/breath.dm (hunk 2 above)
Comment (Rohesie) [reply code/modules/clothing/masks/breath.dm]: Well, the question is on if it works like a rebreather. I was assuming it didn't.

File: code/modules/clothing/clothing.dm [hunk 7]
```diff
@@ -114,6 +114,8 @@
 	flags_armor_protection = FACE|EYES
 	sprite_sheets = list("Vox" = 'icons/mob/species/vox/masks.dmi')
 	var/anti_hug = 0
+	var/toggleable = 0
```
Comment (Rohesie) [inline code/modules/clothing/clothing.dm]:
```suggestion
	var/toggleable = FALSE
```

File: code/modules/clothing/clothing.dm [hunk 8]
```diff
@@ -114,6 +114,8 @@
...
+	var/active = 1
```
Comment (Rohesie) [inline code/modules/clothing/clothing.dm]:
```suggestion
	var/active = TRUE
```

File: code/modules/clothing/masks/breath.dm [hunk 9]
```diff
@@ -62,7 +62,90 @@
...
+			DISABLE_BITFIELD(flags_inventory, (COVERMOUTH|ALLOWREBREATH))
+			DISABLE_BITFIELD(flags_inv_hide, (HIDEFACE|HIDELOWHAIR))
+			DISABLE_BITFIELD(flags_armor_protection, FACE)
+		else
+			ENABLE_BITFIELD(flags_inventory, (COVERMOUTH|ALLOWREBREATH))
+			ENABLE_BITFIELD(flags_inv_hide, (HIDEFACE|HIDELOWHAIR))
+			ENABLE_BITFIELD(flags_armor_protection, FACE)
+		to_chat(usr, "You [active ? "pull [src] up to cover your face" : "pull [src] off your face"].")
+
+
```
Comment (Rohesie) [inline code/modules/clothing/masks/breath.dm]:
```suggestion
```

File: code/modules/clothing/masks/breath.dm [hunk 10]
```diff
@@ -62,7 +62,90 @@
...
 	flags_cold_protection = HEAD
 	min_cold_protection_temperature = ICE_PLANET_MIN_COLD_PROTECTION_TEMPERATURE
+
+/obj/item/clothing/mask/bandanna
+	name = "Tan bandanna"
+	desc = "A colored, resilient, and insulating cloth to cover your face from the elements. This one is Desert Tan"
+	icon_state = "bandanna"
+	item_state = "bandanna"
+	w_class = WEIGHT_CLASS_SMALL
+	flags_armor_protection = 0
```
Comment (Rohesie) [inline code/modules/clothing/masks/breath.dm]:
```suggestion
	flags_armor_protection = NONE
```

**Comment by spookydonut** [comment]
Better.  Probably doesn't need testing if the sprites look fine.  Just merge this after 24hours and everything fixed

**Comment by Rohesie** [comment]
It's mostly for community feedback. Not that I expect this not to be well-received.
But sure, can be merged in 24 hours if it doesn't start killing kittens.

**Pull request merged**